static/report/js/compare.js

```diff
@@ -0,0 +1,639 @@
+    return manage;
+  }
+
+  function make (data) {
+    var item = templates.find('.reticle').clone(true);
+    var pool = templates.find('.pool').clone(true);
+    var end = data.end || Date.create('now').format(Date.ISO8601_DATETIME);
+    var begin = data.begin || Date.create(end).rewind({weeks: 6}).format(Date.ISO8601_DATETIME);
+    view.find('.ranges').append(item);
+    view.find('.observations').append(pool);
+    var control = slippy(item.find('.timeline'),
+      { controls: item.find('.controls')
+      , begin: begin
+      , end: end
+    } );
+    var reticle =  {dom: item, control: control};
+    var color = colorize(pools.length);
+    var lense = ranger(pool, {color: color, begin: begin, end: end, cache: cache});
+    var display =  { dom: pool, control: lense };
+    reticle.dom.find('.timeline').css('border', '1px solid ' + color);
+    display.dom.css('border-color', color);
+    reticle.dom.find("span, :input").css('color', color);
+    // reticle.dom.find(".x.axis line").('styl', color);
+    var axis = d3.select(reticle.dom.get(0)).selectAll('.x.axis');
+    axis.selectAll('line').style('stroke', color);
+    axis.selectAll('path').style('stroke', color);
+    axis.selectAll('text').style('color', color);
+    return { reticle: reticle, display: display, color: color };
+  }
+
+  function on_refocus (ev, range) {
+    var target = $(ev.target);
+    var pool = pools.filter(function (pool) {
+      var timeline = pool.reticle.dom.find('.timeline');
+      return timeline.is(target);
+    }).pop( );
+    if (pool) {
+      pool.display.control.setRange(range[0], range[1]);
+    }
+  }
+
+  function add_new ( ) {
+    var item = make({ });
+    pools.push(item);
+  }
+
+  manage.add_new = add_new;
+
+  return init( );
+}
+
+function pager (opts) {
+  var query = { };
+  var url = opts.url || '/api/v1/entries.json?find[type]=sgv&find[sgv][$gt]=39&count=500000&';
+  query.begin = opts.begin;
+  query.end = opts.end;
+  var payload = [ ];
+  var days = { };
+  var cache = opts.cache;
+  var DATE_FMT = "{yyyy}-{MM}-{dd}";
+
+  var bisect = d3.bisector(function (d) { return Date.create(d.dateString); });
+  function page ( ) {
+  }
+
+  function refresh (start, end) {
+    // console.log("PAYLOAD?", payload);
+    var q = {
+      start: Date.create(Date.create(start).format(DATE_FMT))
+    , end: Date.create(Date.create(end).format(DATE_FMT))
+    };
+    query.begin = start;
+    query.end = end;
+    var range = d3.time.days(q.start, q.end);
+    days = { };
+    console.log('QUERY FOR', range.length, 'days', q, query);
+    iter_query(range);
+    return;
+    if (payload && payload.length > 0) {
+      console.log("NUM DAYS", range.length);
+      // TODO: soft update, only get deltas against the edges of the
+      // cursor.
+      // start.isBefore
+      // start.isBetween
+      // start.isAfter
+      // end.isBefore
+      // end.isBetween
+      // end.isAfter
+      var holding = {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["holding" is defined but never used](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2715308466/issues/source?bid=2452374&fileBranchId=3345581#l294)